app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt

```diff
@@ -112,19 +112,29 @@ class MessageActivity : AppCompatActivity() {
         }
 
         previewButton.setOnClickListener {
-            previewBadge.setValue(
-                    presenter.convertToPreview(if (!content.text.isEmpty()) content.text.toString() else " "),
-                    marquee.isChecked,
-                    flash.isChecked,
-                    Speed.values()[speed.selectedItemPosition],
-                    Mode.values()[mode.selectedItemPosition]
-            )
+            if (!content.text.isEmpty()) {
+                previewBadge.setValue(
+                        presenter.convertToPreview(content.text.toString()),
+                        marquee.isChecked,
+                        flash.isChecked,
+                        Speed.values()[speed.selectedItemPosition],
+                        Mode.values()[mode.selectedItemPosition]
+                )
+            } else {
+                previewBadge.setValue(
+                        Converters.convertDrawableToLEDHex((drawableRecyclerAdapter.getDefaultItem()).image),
+                        false,
+                        false,
+                        Speed.values()[speed.selectedItemPosition],
+                        Mode.FIXED
+                )
+            }
         }
 
         previewButtonDrawable.setOnClickListener {
             if (drawableRecyclerAdapter.getSelectedItem() != -1)
                 previewBadge.setValue(
-                        Converters.convertDrawableToLEDHex((drawableRecyclerAdapter.getSelectedItem() as DrawableInfo).image) as java.util.ArrayList<String>,
+                        Converters.convertDrawableToLEDHex((drawableRecyclerAdapter.getSelectedItem() as DrawableInfo).image),
```

Comment:
Remove cast

When casting anything, think about it 100 times that why is it needed. Is it actually needed or is it just an implementation error on my or previous developer's side.

Almost always, it will be the latter

Reply:
Here either -1(int) or the DrawableInfo will be returned. Hence without the cast, there will be an error.

Reply:
How can an int or object can return from the same method?

Reply:
OK. Please change the return type from Any to DrawableInfo

As I said earlier "just an implementation error on my or previous developer's side."

Reply:
Done.